src/System.Numerics.Vectors/tests/CommonHashCodeTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Numerics.Tests
+{
+    public static class CommonHashCodeTests
+    {
+        private const int ElementsCount = 100000;
+
+        /// <summary>
+        /// Initial version of hash function in System.Numerics.Vectors.
+        /// </summary>
+        private static int BaseHashCodeCombine(int h1, int h2)
+        {
+            return ((h1 << 5) + h1) ^ h2;
+        }
+
+        private static void CountVector2Collisions(Func<int, Vector2> vectorConstructor, out int baseCollisions,
+            out int collisions)
+        {
+            baseCollisions = Enumerable.Range(0, ElementsCount)
+                .Select(vectorConstructor)
+                .GroupBy(x => BaseHashCodeCombine(x.X.GetHashCode(), x.Y.GetHashCode()))
+                .Where(t => t.Count() > 1)
+                .Select(t => t.ToList().Count)
+                .Sum();
+
+            collisions = Enumerable.Range(0, ElementsCount)
+                .Select(vectorConstructor)
+                .GroupBy(x => x.GetHashCode())
+                .Where(t => t.Count() > 1)
+                .Select(t => t.ToList().Count)
+                .Sum();
+        }
+
+        private static void CountVectorCollisions<T>(Func<int, Vector<T>> vectorConstructor, out int baseCollisions,
+            out int collisions) where T : struct
+        {
+            baseCollisions = Enumerable.Range(0, ElementsCount)
+                .Select(vectorConstructor).
+                GroupBy(x =>
+                {
+                    int hash = 1;
+                    for (int i = 0; i < Vector<T>.Count; i++)
+                    {
+                        hash = BaseHashCodeCombine(hash, x[i].GetHashCode());
+                    }
+                    return hash;
+                })
+                .Where(t => t.Count() > 1)
+                .Select(t => t.ToList().Count)
+                .Sum();
+
+            collisions = Enumerable.Range(0, ElementsCount)
+                .Select(vectorConstructor)
+                .GroupBy(x => x.GetHashCode())
+                .Where(t => t.Count() > 1)
+                .Select(t => t.ToList().Count)
+                .Sum();
+        }
+
+        [Fact]
+        public static void CombineHashCodesVector2Tests()
```

Comment:
1. I Agree. Made changes.
2. I Agree, it would be great. But I think that it is too exhaustive. I'm implemented well-known hash-function.
3. I Agree. Made changes.